### PR TITLE
use_resource_file_like_web

### PR DESCRIPTION
**Pull request by Treasure-u**
## Pull Request checklist

I confirm that the following checklist has been completed.

- [ ] Add/Change **test cases** for the changes.
- [ ] Add/Change the related **documentation**, should also change `docs/source/config.rst` when you change file `default_config.yaml`.
- [ ] (Optional) Add your change to `UPDATING.md` when it is an incompatible change.

## What I do
I noticed that when pydolphinscheduler was working with resources in the resource center, it was not similar to a reference file on a web page, but was reading in the file and adding it to commands. Analyzing the reference file format on the web page, we find that ResourceList in taskDefinitionList stores the id of the Resource, not the name. So I've made some changes here, and I've made some changes to how resources are used.

This is the effect of pydolphinscheduler：
image

This is the effect of the dolphinscheduler website：
image

### CODE REVIEW

File: src/pydolphinscheduler/__init__.py [hunk 1]
```diff
@@ -17,4 +17,4 @@
 
 """Init root of pydolphinscheduler."""
 
-__version__ = "4.1.0-dev"
+__version__ = "4.0.3"
```
Comment (zhongjiajie) [inline src/pydolphinscheduler/__init__.py:20]: please revert this change

File: src/pydolphinscheduler/core/task.py [hunk 2]
```diff
@@ -251,24 +246,17 @@ def timeout_flag(self) -> str:
...
         """Get task define attribute `resource_list`."""
         resources = set()
         for res in self._resource_list:
+
             if type(res) == str:
                 resources.add(
                     Resource(
                         name=res, user_name=self.user_name
-                    ).get_fullname_from_database()
+                    ).get_id_from_database()
```
Comment (zhongjiajie) [inline src/pydolphinscheduler/core/task.py:259]: hi @Treasure-u in the latest dev code, we use resource file full path as identify instead of resource id, so in python-sdk python code we will also use full path.

**Comment by zhongjiajie** [comment]
I check out and find the log in version 4.0.1 we still use the resource id for the task https://github.com/apache/dolphinscheduler-sdk-python/blob/7c3185d1ad91a55dc60ea302881e733bd8c13283/src/pydolphinscheduler/core/task.py#L264-L281 and after version 4.0.1 we use the full name of the resource for the task. 

Now I get the point it is because we cherry-pick an unreleased feature to 4.0.2 and 4.0.3 I will revert them and release the correct version to fix them

but the main branch will keep using the full name of the resource because we are gonging to release the version 3.2.0 and it will use resource full name to target resource in the task

File: src/pydolphinscheduler/core/task.py (hunk 2 above)
Comment (Treasure-u) [reply src/pydolphinscheduler/core/task.py:259]: thanks for your reply, im sorry it's too late to reply you,i know it and with pydolphin can do better and better
